docs/painless/painless-basic-syntax.asciidoc

```diff
@@ -0,0 +1,304 @@
+[[painless-basic-syntax]]
+=== Basic Syntax
+
+[[comments]]
+==== Comments
+
+Painless supports both single-line and multi-line comments. Comments can be
+included anywhere within a script. Use the `//` token anywhere on a line to
+specify a single-line comment. All characters from the `//` token to the end
+of the line are ignored. Use an opening `/*` token and a closing `*/` token
+to specify a multi-line comment. Multi-line comments can start anywhere on a
+line, and all characters in between the `/*` token and `*/` token are ignored.
+
+*Grammar:*
+[source,ANTLR4]
+----
+SINGLE_LINE_COMMENT: '//' .*? [\n\r];
+MULTI_LINE_COMMENT: '/*' .*? '*/';
+----
+
+*Examples:*
+
+Single-line and multi-line comments where `<code>` represents tokens within a
+script.
+
+[source,Java]
+----
+// single-line comment
+
+<code> // single-line comment
+
+/* multi-
+   line
+   comment */
+
+<code> /* multi-line
+          comment */ <code>
+
+<code> /* multi-line comment */ <code>
+----
```

Comment:
I'd probably break this up into separate single and multiline examples. (And multi-line doesn't really need a hyphen.)

Reply:
Done.